middlewares/amp-access-cors.js

```diff
@@ -15,16 +15,75 @@
  */
 "use strict";
 
-/* Enable CORS for all AMP Paywall API requests */
+var url = require('url');
+
+var VALID_ORIGINS = [/.*\.ampproject\.org/g];
+var VALID_SOURCE_ORIGINS = [/localhost/, /rocky-sierra-1919\.herokuapp\.com/g];
+
+/**
+ * Enable CORS for all AMP API requests. More information here: 
+ * https://github.com/ampproject/amphtml/blob/master/spec/amp-cors-requests.md 
+ */
 module.exports = function(req, res, next) {
-  if (req.url.startsWith('/amp-')) {
-    // In practice, Origin should be restricted to a few well-known domains.
-    var requestingOrigin = req.header('Origin');
+  if (req.url.indexOf('/api/') > -1) {
+    // check the origin
+    var requestingOrigin = req.headers['origin'];
     console.log('---- requesting origin: ', requestingOrigin);
-    if (requestingOrigin) {
+    if (isValidOrigin(requestingOrigin)) {
```

Comment:
Yup - didn't add the return as this means testing on a localhost no longer works. Probably should make it more explicit and check CORS only when not running on localhost - or is there a better way?